packages/shared/src/schemas.ts

```diff
@@ -25,15 +25,30 @@ export const ccmMetadataSchema = z.object({
 export type CcmMetadata = z.infer<typeof ccmMetadataSchema>;
 
 export const openSwapDepositChannelSchema = z
-  .object({
-    srcAsset: chainflipAsset,
-    destAsset: chainflipAsset,
-    srcChain: chainflipChain,
-    destChain: chainflipChain,
-    destAddress: z.string(),
-    amount: numericString,
-    ccmMetadata: ccmMetadataSchema.optional(),
-  })
+  .intersection(
+    z.object({
+      destAddress: z.string(),
+      amount: numericString,
+      ccmMetadata: ccmMetadataSchema.optional(),
+    }),
+    z.union([
+      z
+        .object({
```

Comment:
this is the old way right? should we add a comment that we can remove it in the future?